save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt

```diff
@@ -70,9 +74,51 @@ internal class ContestController(
     )
     @ApiResponse(responseCode = "200", description = "Successfully fetched contest by it's name.")
     @ApiResponse(responseCode = "404", description = "Contest with such name was not found.")
-    fun getContestByName(@PathVariable contestName: String): Mono<ContestDto> = justOrNotFound(contestService.findByName(contestName))
+    fun getContestByName(@PathVariable contestName: String): Mono<ContestDto> = getContestOrNotFound(contestName)
         .map { it.toDto() }
 
+    @PostMapping("/featured/add-or-delete")
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
+    @Operation(
+        method = "POST",
+        summary = "Create or delete featured contest.",
+        description = "Mark contest to be featured if it is not marked so yet or unmark otherwise.",
+    )
+    @Parameters(
+        Parameter(name = "contestDto", `in` = ParameterIn.DEFAULT, description = "contest requested for creation", required = true),
+    )
+    @ApiResponse(responseCode = "200", description = "Contest was successfully created.")
+    @ApiResponse(responseCode = "404", description = "Contest with given name is not found.")
+    fun addOrDeleteContestToFeatured(
```

Comment:
sorry if I am missing that, but can we check that superusers do this?

Reply:
There is annotation for this on line 82:
```kotlin
@PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
```